### ISSUE TITLE
State file still exist when it should be removed

### ISSUE THREAD

**Issue by ZhengQ2** [rsyslog/rsyslog#4989]
### Expected behavior
```
testbench: TZ env var not set, setting it to UTC
------------------------------------------------------------
17:31:07[0]  Test: ./imfile-statefile-delete.sh
------------------------------------------------------------
config rstb_274901_af91b9aeVm2W_.conf is:
     1  module(load="../runtime/.libs/imdiag")
     2  global(inputs.timeout.shutdown="60000"
     3         default.action.queue.timeoutshutdown="20000"
     4         default.action.queue.timeoutEnqueue="20000")
     5  # use legacy-style for the following settings so that we can override if needed
     6  $MainmsgQueueTimeoutEnqueue 20000
     7  $MainmsgQueueTimeoutShutdown 10000
     8  $IMDiagListenPortFileName rstb_274901_af91b9aeVm2W.imdiag.port
     9  $IMDiagServerRun 0
    10  $IMDiagAbortTimeout 580
    11
    12  :syslogtag, contains, "rsyslogd"  ./rstb_274901_af91b9aeVm2W.started
    13  ###### end of testbench instrumentation part, test conf follows:
    14
    15  global(workDirectory="rstb_274901_af91b9aeVm2W.spool")
    16  module(load="../runtime/.libs/imfile" mode="inotify" PollingInterval="1")
    17  input(type="imfile" tag="file:" file="./rstb_274901_af91b9aeVm2W.input")
    18
    19  template(name="outfmt" type="string" string="%msg:F,58:2%\n")
    20  if $msg contains "msgnum:" then
    21          action(type="omfile" file=`echo $RSYSLOG_OUT_LOG` template="outfmt")
rsyslogd: NOTE: RSYSLOG_DEBUG_TIMEOUTS_TO_STDERR activated
main Q:Reg: worker start requested, num workers currently 0
main Q:Reg: wrkr start initiated with state 0, num workers now 1
rsyslog debug: main Q:Reg: worker 0x5573611370 started
rsyslog debug: main Q:Reg: started with state 3, num workers now 1
17:31:10[3]  rstb_274901_af91b9aeVm2W:.pid found, pid 363
17:31:10[3]  rsyslogd startup msg seen, pid  363
waiting for file rstb_274901_af91b9aeVm2W.imdiag.port
imdiag port: 43265
wait_file_lines success, have 1000 lines, took 0 seconds, file rstb_274901_af91b9aeVm2W.out.log
Shutting down instance 1
imdiag: wait q_empty: qsize 0 nempty 1
imdiag[43265]: mainqueue empty
rsyslogd debug: info: trying to cooperatively stop input ../runtime/.libs/imdiag, timeout 60000 ms
rsyslogd debug: info: trying to cooperatively stop input imfile, timeout 60000 ms
rsyslog debug: main Q:Reg/w0: enter WrkrExecCleanup
rsyslog debug: 0x5573611500: worker exiting
rsyslog debug: main Q:Reg/w0: thread joined
17:31:12[5]  wait on shutdown of 363
17:31:13[6]  Test ./imfile-statefile-delete.sh SUCCESSFUL (took 6 seconds)
-------------------------------------------------------------------------------
```

### Actual behavior
```
testbench: TZ env var not set, setting it to UTC
------------------------------------------------------------
17:36:34[1]  Test: ./imfile-statefile-delete.sh
------------------------------------------------------------
config rstb_614046_af91b9aetaFY_.conf is:
     1  module(load="../runtime/.libs/imdiag")
     2  global(inputs.timeout.shutdown="60000"
     3         default.action.queue.timeoutshutdown="20000"
     4         default.action.queue.timeoutEnqueue="20000")
     5  # use legacy-style for the following settings so that we can override if needed
     6  $MainmsgQueueTimeoutEnqueue 20000
     7  $MainmsgQueueTimeoutShutdown 10000
     8  $IMDiagListenPortFileName rstb_614046_af91b9aetaFY.imdiag.port
     9  $IMDiagServerRun 0
    10  $IMDiagAbortTimeout 580
    11
    12  :syslogtag, contains, "rsyslogd"  ./rstb_614046_af91b9aetaFY.started
    13  ###### end of testbench instrumentation part, test conf follows:
    14
    15  global(workDirectory="rstb_614046_af91b9aetaFY.spool")
    16  module(load="../runtime/.libs/imfile" mode="inotify" PollingInterval="1")
    17  input(type="imfile" tag="file:" file="./rstb_614046_af91b9aetaFY.input")
    18
    19  template(name="outfmt" type="string" string="%msg:F,58:2%\n")
    20  if $msg contains "msgnum:" then
    21          action(type="omfile" file=`echo $RSYSLOG_OUT_LOG` template="outfmt")
rsyslogd: NOTE: RSYSLOG_DEBUG_TIMEOUTS_TO_STDERR activated
main Q:Reg: worker start requested, num workers currently 0
main Q:Reg: wrkr start initiated with state 0, num workers now 1
rsyslog debug: main Q:Reg: worker 0x5569131370 started
rsyslog debug: main Q:Reg: started with state 3, num workers now 1
17:36:36[3]  rstb_614046_af91b9aetaFY:.pid found, pid 289
17:36:36[3]  rsyslogd startup msg seen, pid  289
waiting for file rstb_614046_af91b9aetaFY.imdiag.port
imdiag port: 40469
wait_file_lines success, have 1000 lines, took 0 seconds, file rstb_614046_af91b9aetaFY.out.log
FAIL: state file still exists when it should have been deleted
spool dir is:
total 4
-rw------- 1 root root 137 Sep 20 17:36 imfile-state:3909:cde89929774cfe15
not reporting failure as RSYSLOG_STATSURL is not set
rsyslog pid file still exists, trying to shutdown...
rsyslogd debug: info: trying to cooperatively stop input ../runtime/.libs/imdiag, timeout 60000 ms
rsyslogd debug: info: trying to cooperatively stop input imfile, timeout 60000 ms
rsyslog debug: main Q:Reg/w0: enter WrkrExecCleanup
rsyslog debug: 0x5569131500: worker exiting
rsyslog debug: main Q:Reg/w0: thread joined
17:41:36[303]  FAIL: Test ./imfile-statefile-delete.sh (took 303 seconds)
```

### Steps to reproduce the behavior
`cd rsyslog/ptest/tests`
`./imfile-statefile-delete.sh`

### Environment
- rsyslog version: 8.2208.0
- platform: QEMU emulator version 2.11.1(Debian 1:2.11+dfsg-1ubuntu7.40)

**Comment by rgerhards** [comment]
Apologies for the very late follow-up. This may no longer help the original reporter, but I am recording the current resolution for people finding this through search while working through the backlog. We do not close reports merely because they are old.

This testbench failure was fixed by commit e8c6cc91a. Delayed imfile state cleanup in inotify mode needs a later filesystem event; the corrected test waits past the deletion delay and generates that event. The test remains registered in the current suite, and later imfile cleanup fixes further hardened state-file deletion.

Closing as fixed. If the current test still fails, please open a new issue with the full test log and spool-directory contents.
